lib/ansible/modules/cloud/amazon/lambda.py

```diff
@@ -257,10 +256,13 @@ def main():
     required_together = [['s3_key', 's3_bucket'],
                          ['vpc_subnet_ids', 'vpc_security_group_ids']]
 
+    required_if = [['state', 'present', ['runtime', 'handler', 'role']]]
```

Comment:
I'd rather not split it, and I'm also against an `update` mode. 

I'm ok with requiring role, and I can't see many reasons someone would want to use the module without one. Maybe if they're *only* updating the zipfile as part of a deploy?